fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/TransactionalFcrepoClient.java

```diff
@@ -134,7 +148,11 @@ private URI rewriteUri(URI rescUri) {
         URI txUri = FedoraTransaction.txUriThread.get();
         String rescId = rescUri.toString();
         int txIdIndex = rescId.indexOf(REST);
-        rescId = rescId.substring(txIdIndex + REST.length());
+        int rescPathIndex = txIdIndex + REST.length();
```

Comment:
i'm not sure i completely follow what's going on in lines 151-155. maybe a comment would be helpful. if i'm understanding correctly, if there's something left in the uri beyond the txid itself, these lines will join that something to the txUri?